File: front-end/src/module/model/server/instituicao-curso-ocorrencia-periodo-aluno.ts
import { InstituicaoCursoTurma } from './instituicao-curso-turma';
import { InstituicaoCursoPeriodo } from './instituicao-curso-periodo';
import { InstituicaoCursoOcorrenciaAluno } from './instituicao-curso-ocorrencia-aluno';
import { Aluno } from './aluno';

export class InstituicaoCursoOcorrenciaPeriodoAluno {
    id?: string;
    aluno: Aluno;
    instituicaoCursoTurma: InstituicaoCursoTurma;
    instituicaoCursoPeriodo: InstituicaoCursoPeriodo;
}

File: back-end/Domain/src/Models/Instituicao/InstituicaoCursoOcorrenciaPeriodoAluno.cs
using System;
using System.ComponentModel.DataAnnotations;
using Domain.Models.Interface;

namespace Domain.Models {
    public class InstituicaoCursoOcorrenciaPeriodoAluno : IBaseModel {

        [Key]
        public long ID { get; set; }
        public InstituicaoCursoOcorrenciaPeriodo InstituicaoCursoOcorrenciaPeriodo { get; set; }
        public InstituicaoCursoOcorrenciaAluno InstituicaoCursoOcorrenciaAluno { get; set; }
        public InstituicaoCursoPeriodo InstituicaoCursoPeriodo { get; set; }
        public InstituicaoCursoTurma InstituicaoCursoTurma { get; set; }
        public bool Confirmado { get; set; }
        public DateTime? Ativo { get; set; } = null;

    }

}